app/Http/Controllers/PosgradController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Arr;
+use Illuminate\Validation\Rule;
+use Uspdev\Replicado\Posgraduacao;
+use Uspdev\Replicado\DB;
+
+class PosgradController extends Controller
+{
+    public function index(Request $request)
+    {
+        $this->authorize('admin');
+
+        # tratando POST
+        if (isset($request->codcur)) {
+            if ($request->codcur != 0) {
+                $request->validate([
+                    'codcur' => ['integer', Rule::in(Arr::pluck($programas, 'codcur'))],
+                ]);
+            }
+            session(['codcur' => $request->codcur]);
+        }
+
+        $programas = SELF::listarProgramas();
```

Comment:
Eu já aceitei. Mas essa linha 
`$programas = SELF::listarProgramas();` não deveria estar antes de:
`
'codcur' => ['integer', Rule::in(Arr::pluck($programas, 'codcur'))],` ?